internal/server/worker.go

```diff
@@ -12,12 +12,18 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-type WorkerType string
+type (
+	WorkerType       string
+	OperationalState string
```

Comment:
Consider making this a different type that can be converted to a string when needed. The reason is that you are using an *atomic.Value to store it, but this requires runtime type assertions. If it was either a uint32 or a struct with a value then you could use `atomic.Uint32` or `atomic.Pointer` for compile-time type safety.